heaps.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 The Matrix.org Foundation C.I.C.
```

Comment:
Naming: Please add `stateresolutionv2` to this filename, as otherwise it's not obvious that this is about it without opening the file.